Unauthorized error 401 with Gemini

I am trying a simple version of the example in the documentation but using Gemini instead of OpenAI.
The code is in Kotlin.

```
@RegisterAiService
interface MyAiService {

    @SystemMessage("You are a professional poet")
    @UserMessage(
        """
                Write a poem about {topic}. The poem should be {lines} lines long. 
            
            """
    )
    fun writeAPoem(topic: String, lines: Int): String?

}
```

```
@ApplicationScoped
class Main {

    @Inject @Default
    private lateinit var aiService: MyAiService

    @Startup
    fun start() {
        aiService.writeAPoem("freedom", 5)
    }
}
```

application.properties

> quarkus.langchain4j.log-requests=true
> quarkus.langchain4j.log-responses=true
> quarkus.langchain4j.vertexai.gemini.location=us-central1
> quarkus.langchain4j.vertexai.gemini.project-id=my-project

Before I run from the IDE I executed this command in the shell:

`gcloud auth application-default login
`

Console log:

> 2024-08-26 16:45:43,876 INFO  [io.qua.lan.ver.run.gem.VertxAiGeminiRestApi$VertxAiClientLogger] (vert.x-eventloop-thread-2) Request:
> - method: POST
> - url: https://us-central1-aiplatform.googleapis.com/v1/projects/my-project/locations/us-central1/publishers/google/models/gemini-pro:generateContent
> - headers: [Authorization: ya29.a0AcM612y_...UH1A0175], [Content-Type: application/json], [User-Agent: Quarkus REST Client], [content-length: 274]
> - body: {"contents":[{"role":"user","parts":[{"text":"\n                Write a poem about freedom. The poem should be 5 lines long. \n            \n            "}]}],"systemInstruction":{"parts":[{"text":"You are a professional poet"}]},"generationConfig":{"maxOutputTokens":8192}}
> 2024-08-26 16:45:44,682 INFO  [io.qua.lan.ver.run.gem.VertxAiGeminiRestApi$VertxAiClientLogger] (vert.x-eventloop-thread-2) Response:
> - status code: 401
> - headers: [WWW-Authenticate: Bearer realm="https://accounts.google.com/"], [Vary: X-Origin], [Vary: Referer], [Content-Type: application/json; charset=UTF-8], [Date: Mon, 26 Aug 2024 13:45:44 GMT], [Server: scaffolding on HTTPServer2], [Cache-Control: private], [X-XSS-Protection: 0], [X-Frame-Options: SAMEORIGIN], [X-Content-Type-Options: nosniff], [Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000], [Accept-Ranges: none], [Vary: Origin,Accept-Encoding], [Transfer-Encoding: chunked]
> - body: {
>   "error": {
>     "code": 401,
>     "message": "Request is missing required authentication credential. Expected OAuth 2 access token, login cookie or other valid authentication credential. See https://developers.google.com/identity/sign-in/web/devconsole-project.",
>     "status": "UNAUTHENTICATED",
>     "details": [
>       {
>         "@type": "type.googleapis.com/google.rpc.ErrorInfo",
>         "reason": "CREDENTIALS_MISSING",
>         "domain": "googleapis.com",
>         "metadata": {
>           "method": "google.cloud.aiplatform.v1.PredictionService.GenerateContent",
>           "service": "aiplatform.googleapis.com"
>         }
>       }
>     ]
>   }
> }
> 
> 2024-08-26 16:45:44,686 ERROR [io.qua.lan.run.ais.AiServiceMethodImplementationSupport] (Quarkus Main Thread) Execution of com.example.MyAiService#writeAPoem failed: org.jboss.resteasy.reactive.ClientWebApplicationException: Received: 'Unauthorized, status code 401' when invoking: Rest Client method: 'io.quarkiverse.langchain4j.vertexai.runtime.gemini.VertxAiGeminiRestApi#generateContent'
> 	at org.jboss.resteasy.reactive.client.impl.RestClientRequestContext.unwrapException(RestClientRequestContext.java:195)
> 	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.handleException(AbstractResteasyReactiveContext.java:329)
> 	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:175)
> 	at io.smallrye.context.impl.wrappers.SlowContextualRunnable.run(SlowContextualRunnable.java:19)
> 	at org.jboss.resteasy.reactive.client.handlers.ClientSwitchToRequestContextRestHandler$1$1.handle(ClientSwitchToRequestContextRestHandler.java:38)
> 	at org.jboss.resteasy.reactive.client.handlers.ClientSwitchToRequestContextRestHandler$1$1.handle(ClientSwitchToRequestContextRestHandler.java:35)
> 	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:279)
> 	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:261)
> 	at io.vertx.core.impl.ContextInternal.lambda$runOnContext$0(ContextInternal.java:59)
> 	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173)
> 	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166)
> 	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:469)
> 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:566)
> 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:994)
> 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
> 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
> 	at java.base/java.lang.Thread.run(Thread.java:1583)
> Caused by: jakarta.ws.rs.WebApplicationException: Unauthorized, status code 401
> 	at io.quarkus.rest.client.reactive.runtime.DefaultMicroprofileRestClientExceptionMapper.toThrowable(DefaultMicroprofileRestClientExceptionMapper.java:19)
> 	at io.quarkus.rest.client.reactive.runtime.MicroProfileRestClientResponseFilter.filter(MicroProfileRestClientResponseFilter.java:52)
> 	at org.jboss.resteasy.reactive.client.handlers.ClientResponseFilterRestHandler.handle(ClientResponseFilterRestHandler.java:21)
> 	at org.jboss.resteasy.reactive.client.handlers.ClientResponseFilterRestHandler.handle(ClientResponseFilterRestHandler.java:10)
> 	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.invokeHandler(AbstractResteasyReactiveContext.java:231)
> 	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
> 	... 14 more
> 
> 2024-08-26 16:45:44,695 ERROR [io.qua.run.Application] (Quarkus Main Thread) Failed to start application: java.lang.RuntimeException: Failed to start quarkus
> 	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
> 	at io.quarkus.runtime.Application.start(Application.java:101)
> 	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:119)
> 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
> 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
> 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
> 	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
> 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
> 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
> 	at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:116)
> 	at java.base/java.lang.Thread.run(Thread.java:1583)
> Caused by: org.jboss.resteasy.reactive.ClientWebApplicationException: Received: 'Unauthorized, status code 401' when invoking: Rest Client method: 'io.quarkiverse.langchain4j.vertexai.runtime.gemini.VertxAiGeminiRestApi#generateContent'
> 	at org.jboss.resteasy.reactive.client.impl.RestClientRequestContext.unwrapException(RestClientRequestContext.java:195)
> 	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.handleException(AbstractResteasyReactiveContext.java:329)
> 	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:175)
> 	at io.smallrye.context.impl.wrappers.SlowContextualRunnable.run(SlowContextualRunnable.java:19)
> 	at org.jboss.resteasy.reactive.client.handlers.ClientSwitchToRequestContextRestHandler$1$1.handle(ClientSwitchToRequestContextRestHandler.java:38)
> 	at org.jboss.resteasy.reactive.client.handlers.ClientSwitchToRequestContextRestHandler$1$1.handle(ClientSwitchToRequestContextRestHandler.java:35)
> 	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:279)
> 	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:261)
> 	at io.vertx.core.impl.ContextInternal.lambda$runOnContext$0(ContextInternal.java:59)
> 	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173)
> 	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166)
> 	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:469)
> 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:566)
> 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:994)
> 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
> 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
> 	... 1 more
> Caused by: jakarta.ws.rs.WebApplicationException: Unauthorized, status code 401
> 	at io.quarkus.rest.client.reactive.runtime.DefaultMicroprofileRestClientExceptionMapper.toThrowable(DefaultMicroprofileRestClientExceptionMapper.java:19)
> 	at io.quarkus.rest.client.reactive.runtime.MicroProfileRestClientResponseFilter.filter(MicroProfileRestClientResponseFilter.java:52)
> 	at org.jboss.resteasy.reactive.client.handlers.ClientResponseFilterRestHandler.handle(ClientResponseFilterRestHandler.java:21)
> 	at org.jboss.resteasy.reactive.client.handlers.ClientResponseFilterRestHandler.handle(ClientResponseFilterRestHandler.java:10)
> 	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.invokeHandler(AbstractResteasyReactiveContext.java:231)
> 	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
> 	... 14 more

--- a/model-providers/vertex-ai-gemini/runtime/src/main/java/io/quarkiverse/langchain4j/vertexai/runtime/gemini/VertxAiGeminiRestApi.java
+++ b/model-providers/vertex-ai-gemini/runtime/src/main/java/io/quarkiverse/langchain4j/vertexai/runtime/gemini/VertxAiGeminiRestApi.java
@@ -115,7 +115,7 @@ public interface VertxAiGeminiRestApi {
             try {
                 var credentials = GoogleCredentials.getApplicationDefault();
                 credentials.refreshIfExpired();
-                return credentials.getAccessToken().getTokenValue();
+                return "Bearer " + credentials.getAccessToken().getTokenValue();
             } catch (IOException e) {
                 throw new UncheckedIOException(e);
             }

